src/libraries/System.Net.Security/src/System/Net/NegotiateAuthenticationPal.Unix.cs

```diff
@@ -0,0 +1,803 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Buffers.Binary;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Security.Principal;
+using System.Security.Authentication.ExtendedProtection;
+using System.Text;
+using System.Net.Security;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Net
+{
+    internal partial class NegotiateAuthenticationPal
+    {
+        private static bool _useManagedNtlm;
+        private static bool _isGssApiAvailable;
+
+#pragma warning disable CA1810 // explicit static cctor
+        static NegotiateAuthenticationPal()
+        {
+            try
+            {
+                if (!Interop.NetSecurityNative.IsNtlmInstalled())
+                {
+                    _useManagedNtlm = !AppContext.TryGetSwitch("System.Net.Security.UseManagedNtlm", out bool useManagedNtlm) || useManagedNtlm;
+                }
+                else
+                {
+                    _useManagedNtlm = AppContext.TryGetSwitch("System.Net.Security.UseManagedNtlm", out bool useManagedNtlm) && useManagedNtlm;
+                }
+                _isGssApiAvailable = true;
+            }
+            catch (EntryPointNotFoundException)
+            {
+                // GSSAPI shim may not be available on some platforms (Linux Bionic)
+                _isGssApiAvailable = false;
+            }
```

Comment:
FWIW I am not entirely keen on shipping the fallbacks this way. 

I propose to change it to the following:
- Add the optional `System.Net.Security.UseManagedNtlm` and default to `false`.
- Do not fallback to managed NTLM/SPNEGO if the native GSSAPI is missing NTLM-SSP (ie. align behavior with .NET 7 and older).
- Add ILLink script that links away `ManagedNtlm` and `ManagedSpnego` unless `System.Net.Security.UseManagedNtlm` is set.
- Update NativeAOT linux-bionic builds to set `System.Net.Security.UseManagedNtlm=true` unconditionally.

Reply:
Would it make sense to fallback to managed only for NTLM and not Negotiate if NTLM-SSP is missing? We would not interfere with negotiation in any way but it would provide some legacy capability

Reply:
For reference, this is implementation of the proposal above: https://github.com/dotnet/runtime/commit/aa22b0a369e4910b86e6d5492195d4b51b9e996f
It mostly reverts all the fallback paths to .NET 7 behavior and only enables them with the AppContext switch.

Doing only managed NTLM without managed SPNEGO is feasible but not very practical. HTTP servers almost universally send either Negotiate only, or both Negotiate+NTLM. In both cases this would still fail the authentication because Negotiate gets the preference. There may be some SMTP server that supports only NTLM but that's really niche use case.

Reply:
That said, if you feel like we should do `ManagedNtlm`-only and not `ManagedSpNego` for `!Interop.NetSecurityNative.IsNtlmInstalled()` I can do it. It will just need a re-run of the whole test matrix to figure out what breaks.

Reply:
I was thinking about  https://github.com/dotnet/runtime/issues/87568 and perhaps Azure functions were there is no control over installed packages. But I guess the AppContext should be good enough for now.

Reply:
I pushed the one extra commit mentioned above. It passed CI in #89051 test PR except for few legs that failed due to infra failures. They are now re-running 🤞